lib/response.js

```diff
@@ -464,33 +466,20 @@ function patch(Response) {
             type = mime.lookup(type);
         }
 
-        // If we were unable to derive a valid type, default to treating it as
-        // arbitrary binary data per RFC 2046 Section 4.5.1
-        if (!self.formatters[type] && self.acceptable.indexOf(type) === -1) {
-            type = 'application/octet-stream';
-        }
-
         formatter = self.formatters[type] || self.formatters['*/*'];
 
-        // If after the above attempts we were still unable to derive a
-        // formatter, provide a meaningful error message
-        if (!formatter) {
-            return formatterError(
-                self,
-                new errors.InternalServerError({
-                    message:
-                        'could not find formatter for application/octet-stream'
-                })
-            );
-        }
-
         if (self._charSet) {
             type = type + '; charset=' + self._charSet;
         }
 
-        // Update header to the derived content type for our formatter
+        // Update Content-Type header to the one originally set or to the type
+        // inferred from the most relevant formatter found.
         self.setHeader('Content-Type', type);
 
+        if (!formatter) {
+            return flush(self, body);
```

Comment:
I was wondering if we should assert on this unformatted body to ensure it is either string/buffer, but I realized we call `flush()` in many places so it may be outside of the scope of this PR. I would expect res.write to throw in that scenario, but it may be better if we can catch it earlier and provide a better error message.

Reply:
That's a good point. We're already [asserting on that in the `sendRaw` case](https://github.com/restify/node-restify/blob/9a59b0f8ce6b4c75b3904aff00d11e60b40a6846/lib/response.js#L422-L428), so we might as well do the same as you suggest in this case.